zcash_extras/src/wallet.rs

```diff
@@ -209,6 +247,11 @@ where
         }
     };
 
+    // Automatically decrypt and store any outputs sent to our wallet, including change.
+    // This uses our viewing keys to find any outputs we can decrypt, creates decrypted
+    // note data for spendability, and saves them to the wallet database.
+    decrypt_and_store_transaction(params, wallet_db, &tx).await?;
+
     wallet_db
         .store_sent_tx(&SentTransaction {
```

Comment:
Looks like this store_sent_tx call stores the created transaction output in the walletdb.
Wouldn't the decrypt_and_store_transaction call do a similar thing?

Reply:
`store_sent_tx` - stores notes we created to send value to others, while `decrypt_and_store_transaction` calls `store_received_notes`(stores notes we can spend) internally

Reply:
Aha, we need to add the change as a received output, to fix the balance